### DISCUSSION TITLE
How would you create a random Mac UUID?

### DISCUSSION THREAD

**Discussion by CKingX** [uuid-rs/uuid#608] [Q&A]
I see a Version::Mac and UUID Builder, but you would have to provide the bytes. Is it possible to use say new_v1() (to get the UUID and convert it to bytes and use those bytes in the builder) or do we get random bytes ourselves and then pass it to the Builder?

**Comment by KodrAus** [comment]
Hi @CKingX :wave:

You should be able to use a library like `rand` to generate a random `[u8; 6]` array, and then pass that to [`Uuid::new_v1`](https://docs.rs/uuid/latest/uuid/struct.Uuid.html#method.new_v1). It kind of sounds like you might really be after is [a V7 UUID](https://github.com/uuid-rs/uuid/issues/523), but we don't support those yet 🙂

**Comment by CKingX** [comment]
Thank you! I was just wondering how to create any of these UUIDs for ruuhid:
* Mac
* Dce
* Md5

Since the only reference to Mac UUID was in the Version enum, and Version enum itself mentioned Mac is a version 1 UUID, I thought it would be a version 1 UUID. I will be reading up more on the different kinds of UUIDs

**Reply by KodrAus** [reply]
I think the `ruuhid` utility is still mostly stubbed at this stage, so it only supports generating random (v4) UUIDs. The `uuid` library itself has features and methods for:

- v1: MAC
- v3: MD5
- v4: Random
- v5: SHA1

The notable omission there is v2 (DCE) support since RFC4122 doesn't actually specify how to generate them.
